lib/cisco_node_utils/itd_device_group_node.rb

```diff
@@ -0,0 +1,145 @@
+      fail "itd device-group #{itd_dg_name} does not exist" if
+      @itddg.nil?
+      @name = node_name
+      @node_type = node_type
+
+      set_args_keys_default
+      create_node if instantiate
+    end
+
+    # itd_device_group_nodes have the name form as
+    # node ip 1.1.1.1
+    # node IPv6 2000::1
+    # and they depdend on the device_group
+    def self.itd_nodes(node_name=nil)
+      fail TypeError unless node_name.is_a?(String) || node_name.nil?
+      itd_nodes = {}
+      itd_list = ItdDeviceGroup.itds
+      return itd_nodes if itd_list.nil?
+      itd_list.keys.each do |name|
+        itd_nodes[name] = {}
+        match = config_get('itd_device_group',
+                           'all_itd_device_group_nodes', name: name)
+        next if match.nil?
+        match.each do |line|
+          local = line.split
+          ntype = local[0]
+          nname = local[1].strip
+          next unless node_name.nil? || nname == node_name
+          itd_nodes[name][nname] =
+              ItdDeviceGroupNode.new(name, nname, ntype, false)
+        end
+      end
+      itd_nodes
+    end
+
+    ########################################################
+    #                      PROPERTIES                      #
+    ########################################################
+
+    def ==(other)
+      (itd_device_group_name == other.itd_device_group_name) &&
+        (name == other.name) && (node_type == other.node_type)
+    end
+
+    def create_node
+      config_set('itd_device_group', 'create_node',
+                 name: @itddg.name, ntype: @node_type, nname: @name)
+    end
+
+    def destroy
+      config_set('itd_device_group', 'destroy_node', name: @itddg.name,
+                ntype: @node_type, nname: @name)
+    end
+
+    # Helper method to delete @set_args hash keys
+    def set_args_keys_default
+      @set_args = { name: @itddg.name }
+      @set_args[:ntype] = @node_type
+      @set_args[:nname] = @name
+      @get_args = @set_args
+    end
+
+    def hot_standby
+      config_get('itd_device_group', 'hot_standby', @get_args)
+    end
+
+    # DO NOT call this directly
+    def lhot_standby=(state)
+      no_cmd = (state ? '' : 'no')
+      @set_args[:state] = no_cmd
+      config_set('itd_device_group', 'hot_standby', @set_args)
+      set_args_keys_default
+    end
+
+    def default_hot_standby
+      config_get_default('itd_device_group', 'hot_standby')
+    end
+
+    def weight
+      config_get('itd_device_group', 'weight', @get_args)
+    end
+
+    # DO NOT call this directly
+    def lweight=(val)
+      @set_args[:state] = val == default_weight ? 'no' : ''
+      @set_args[:weight] = val
+      config_set('itd_device_group', 'weight', @set_args)
+    end
+
+    # Call this for setting hot_standby and weight together because
+    # the CLI is pretty weird and it accepts these params in a very
+    # particular way and they cannot even be reset unless proper
+    # order is followed
+    def hs_weight=(hs, wt)
```

Comment:
This I am not in complete agreement with you. 
If you look at 
https://github.com/saichint/cisco-network-node-utils/blob/develop/lib/cisco_node_utils/fabricpath_global.rb#L323
it is following different approach. Also, fabricpath code came later than ospf, I believe. So I don't think it is consistent everywhere. 
If you insist, I will change it but I am not going to agree that this will make consistent code, it would be just an opinion difference.